src/gemdat/transitions.py

```diff
@@ -41,43 +42,63 @@ def __init__(
         self,
         *,
         trajectory: Trajectory,
-        structure: Structure,
+        sites: Structure,
+        dist_close: float,
         events: pd.DataFrame,
         states: np.ndarray,
         inner_states: np.ndarray,
-        dist_close: float,
     ):
         """Store event data for jumps and transitions between sites.
 
         Parameters
         ----------
+        trajectory : Trajectory
+            Trajectory of species of interest (e.g. diffusing)
+            for which transitions are generated
+        sites : Structure
+            Structure with known sites used for calculation of events
+        dist_close: float
+            Custom diameter of all sites
         events : np.ndarray
             Input events
         states : np.ndarray
             Input states
         inner_states : np.ndarray
             Input states for inner sites
-        structure : Structure
-            Structure with know sites used for calculation of events
-        dist_close: float
-            Custom diameter of all sites
-        trajectory : Trajectory
-            Trajectory of species of interest (e.g. diffusing)
-            for which transitions are generated
         """
-        self.structure = structure
+        self.sites = sites
         self.trajectory = trajectory
         self.dist_close = dist_close
         self.states = states
         self.inner_states = inner_states
         self.events = events
 
+    @property
+    def n_floating(self) -> int:
+        """Return number of floating species."""
+        return len(self.trajectory.species)
+
+    @property
+    def n_states(self) -> int:
+        """Return number of states."""
+        return len(self.states)
+
+    @property
+    def n_events(self) -> int:
+        """Return number of events."""
+        return len(self.events)
+
+    @property
+    def n_sites(self) -> int:
+        """Return number of sites."""
+        return len(self.sites)
+
     @classmethod
     def from_trajectory(
         cls,
         *,
         trajectory: Trajectory,
-        structure: Structure,
+        sites: Structure,
```

Comment:
why this change from structure -> sites, while the class is still called Structure, that is a bit confusing to me ...

Reply:
I wanted to make clear that this is not an arbitrary structure, but the collection of known sites that the user inputs. The goal is to make it easier to track what this is. In many places we already used `sites` (previously an instance of SitesData), which now refers to a Structure. So there it is maybe a bit less confusing.